compiler/lib/deadcode.ml

```diff
@@ -319,6 +344,8 @@ let f ({ blocks; _ } as p : Code.program) =
             })
       blocks
   in
+  let p = { p with blocks } in
+  let p = remove_empty_blocks ~live_vars:st.live p in
   if times () then Format.eprintf "  dead code elim.: %a@." Timer.print t;
```

Comment:
at line 323, we have 
```
if not (Addr.Set.mem pc st.reachable_blocks)
        then (
          st.deleted_blocks <- st.deleted_blocks + 1;
          None)
```
Maybe we could remove this logic (and turn the `filter_map` into a `map`) and let remove_unused_blocks do its job.

Reply:
It seems better to remove this dead code early